### ISSUE TITLE
Refactoring and specs not merged up

### ISSUE THREAD

**Issue by soumyaray** [Lemon555/NEO#4]
I see your team has a branch with refactoring (including isolation of api and lib code), and a branch with specs... but these are not merged so i can't tell if the refactored code passes the tests 😆 
See your commit structure here: [https://github.com/Lemon555/NEO/network](https://github.com/Lemon555/NEO/network)

**Comment by marandy0905** [comment]
@soumyaray Actually, it doesn't. The test only pass the api_library one.

**Comment by soumyaray** [comment]
Right, that's why tests are so important to write early on (before refactoring), so that all branches can test their code as they develop :)
